tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/SymbStateDecoder.scala

```diff
@@ -212,6 +156,61 @@ class SymbStateDecoder(solverContext: SolverContext, rewriter: SymbStateRewriter
       throw new RewriterException("Don't know how to decode the cell %s of type %s".format(cell, cell.cellType), NullEx)
   }
 
+  private def decodeFunToTlaEx(arena: Arena, cell: ArenaCell, funT1: FunT1): TlaEx = {
+    // a function is represented with the relation {(x, f[x]) : x \in S}
+    val relation = arena.getCdm(cell)
+
+    def isInRelation(pair: ArenaCell): Boolean = {
```

Comment:
You can pattern match `funT1` at the start for readability: 
```suggestion
  private def decodeFunToTlaEx(arena: Arena, cell: ArenaCell, funT1: FunT1): TlaEx = {
    // a function is represented with the relation {(x, f[x]) : x \in S}
    val relation = arena.getCdm(cell)
    val FunT1(domT, cdmT) = funT1

    def isInRelation(pair: ArenaCell): Boolean = {
```